redis_cache/backends/base.py

```diff
@@ -435,34 +437,42 @@ def get_or_set(
         lock_key = "__lock__" + key
         fresh_key = "__fresh__" + key
 
-        is_fresh = self._get(client, fresh_key)
         value = self._get(client, key)
-
-        if is_fresh:
+        is_fresh = self._get(client, fresh_key)
+        if value is not None and is_fresh:
```

Comment:
```suggestion
        if is_fresh:
```
I suggest maybe we leave the None check out of here because `func` could explicitly return None here